libraries/stanza_filter.rb

```diff
@@ -0,0 +1,134 @@
+#
+# Copyright 2018 Sous Chefs
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Filter insert or change keys in files formatted as stanzas
+module Line
+  class Filter
+    def stanza(current, args)
+      # Assumes stanzas are named uniquely across the file and contiguous
+      # Stanza starts with ^[<name>]$
+      # Stanza ends with next stanza or EOF
+      # Sets one instance of a key in a stanza to a new value TODO: set last found
```

Comment:
Is this todo still valid? needs an issue?

Reply:
Changing to WIP.  Can't get to it today.

Reply:
No worries - just wanted to make sure it didn't get lost

Reply:
This PR should be ready now.  Thanks for finding those TODOs.  I added tests to verify the behavior.